processing/src/main/java/io/druid/query/ordering/StringComparators.java

```diff
@@ -0,0 +1,311 @@
+        {
+          return 1;
+        } else if (delta == 0 && ch0 != ch1)
+        {
+          delta = valueOf(ch0) - valueOf(ch1);
+        }
+
+        if (pos[0] < str0.length())
+        {
+          ch0 = str0.codePointAt(pos[0]);
+          if (isDigit(ch0))
+          {
+            pos[0] += Character.charCount(ch0);
+          } else
+          {
+            ch0 = -1;
+          }
+        } else
+        {
+          ch0 = -1;
+        }
+
+        if (pos[1] < str1.length())
+        {
+          ch1 = str1.codePointAt(pos[1]);
+          if (isDigit(ch1))
+          {
+            pos[1] += Character.charCount(ch1);
+          } else
+          {
+            ch1 = -1;
+          }
+        } else
+        {
+          ch1 = -1;
+        }
+      }
+    }
+
+    private boolean isDigit(int ch)
+    {
+      return (ch >= '0' && ch <= '9') ||
+          (ch >= '\u0660' && ch <= '\u0669') ||
+          (ch >= '\u06F0' && ch <= '\u06F9') ||
+          (ch >= '\u0966' && ch <= '\u096F') ||
+          (ch >= '\uFF10' && ch <= '\uFF19');
+    }
+
+    private boolean isZero(int ch)
+    {
+      return ch == '0' || ch == '\u0660' || ch == '\u06F0' || ch == '\u0966' || ch == '\uFF10';
+    }
+
+    private int valueOf(int digit)
+    {
+      if (digit <= '9')
+      {
+        return digit - '0';
+      }
+      if (digit <= '\u0669')
+      {
+        return digit - '\u0660';
+      }
+      if (digit <= '\u06F9')
+      {
+        return digit - '\u06F0';
+      }
+      if (digit <= '\u096F')
+      {
+        return digit - '\u0966';
+      }
+      if (digit <= '\uFF19')
+      {
+        return digit - '\uFF10';
+      }
+
+      return digit;
+    }
+
+    private int compareNonNumeric(String str0, String str1, int[] pos)
+    {
+      // find the end of both non-numeric substrings
+      int start0 = pos[0];
+      int ch0 = str0.codePointAt(pos[0]);
+      pos[0] += Character.charCount(ch0);
+      while (pos[0] < str0.length() && !isDigit(ch0 = str0.codePointAt(pos[0])))
+      {
+        pos[0] += Character.charCount(ch0);
+      }
+
+      int start1 = pos[1];
+      int ch1 = str1.codePointAt(pos[1]);
+      pos[1] += Character.charCount(ch1);
+      while (pos[1] < str1.length() && !isDigit(ch1 = str1.codePointAt(pos[1])))
+      {
+        pos[1] += Character.charCount(ch1);
+      }
+
+      // compare the substrings
+      return String.CASE_INSENSITIVE_ORDER.compare(str0.substring(start0, pos[0]), str1.substring(start1, pos[1]));
+    }
+    
+    @Override
+    public boolean equals(Object o)
+    {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      
+      return true;
+    }
+    
+    @Override
+    public String toString()
+    {
+      return StringComparators.ALPHANUMERIC;
+    }
+  }
+
+  public static StringComparator makeComparator(String type)
+  {
+    if (type.equals(StringComparators.LEXICOGRAPHIC)) {
```

Comment:
StringComparator objects is set to be deserialized as below by https://github.com/druid-io/druid/pull/2393/files#diff-7f2d04e619ed92b405836719521f2a44R46

```
{"queryType":"groupBy","dataSource":{"type":"table","name":"testing"},"intervals":{"type":"intervals","intervals":["2011-04-01T00:00:00.000Z/2011-04-03T00:00:00.000Z"]},"filter":null,"granularity":
....
[{"type":"fieldAccess","name":"x","fieldName":"idx"}],"having":null,"limitSpec":{"type":"default","columns":[{"dimension":"alias","direction":"ASCENDING","dimensionComparator":{"type":"lexicographic"}}],"limit":100},"context":null,"descending":false}
```
I think [OrderByColumnSpec.create(Object obj)](https://github.com/druid-io/druid/pull/2393/files#diff-d970a21c07a25a78efc6ce7421f8e213L66) should be changed to make a full use of polymorphic deserialization. It would be better making another PR for it. Am I right?